### PR TITLE
[DTensor] Add aten.any.default,dim,out to linear_reduction_strategy

### PR DESCRIPTION
**Pull request by wz337**
For `aten.any`, we can use `reduce_op="sum"` as the linear reduction op. 

When we do `all_reduce` with `reduce_op="sum"` on bool tensor, if one rank returns `torch.Tensor([True]) `, then the reduction result is `torch.Tensor([True]) `. Only when all ranks return `torch.Tensor([False]) ` would the reduction result be `torch.Tensor([False]) `. This matches with `any`'s behavior.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wconstab @d4l3k @c-p-i-o @tianyu-l

### CODE REVIEW

File: test/distributed/_tensor/test_math_ops.py
Comment (tianyu-l) [inline test/distributed/_tensor/test_math_ops.py:30]: Alternatively you can use `torch.randn(8, 2).bool()` here.

Also I think for `aten.all` the conversion is implicit? so we probably don't need this if-else anyway. I suggest we either put `all` and `any` together into this branch, or remove this branch.

**Review by tianyu-l — APPROVED** [review]
thanks! pls address inline comments before merging.

**Review by chuanhaozhuge — APPROVED** [review]
thanks Iris!

File: test/distributed/_tensor/test_math_ops.py
Comment (wz337) [reply test/distributed/_tensor/test_math_ops.py]: The only number that would be evaluated to be False is 0 so most likely we might not have any False in `torch.randn(8, 2).bool()`. Keeping the < 0 here so we would have some False in the bool tensor. 

I'll move "all" to the branch as well.

Comment (wz337) [reply test/distributed/_tensor/test_math_ops.py]: Actually, I realized the result for `all` might be wrong, but the test didn't catch it since we are testing it on `tensor = torch.randn(12, 8, 8) `, which will likely have everything to be True. 

Added a TODO to check `all` correctness but want to land this to unblock something else first.

**Pull request closed without merging**
